### PR TITLE
feat(wallet-dashboard): add settings menu and network switcher

### LINKED ISSUE #4489: Add network switcher

### PR DESCRIPTION
**Pull request by brancoder**
# Description of change

closes #4489

### CODE REVIEW

File: apps/wallet-dashboard/components/Dialogs/settings/hooks/useSettingsDialog.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { useState } from 'react';
+import { SettingsDialogView } from '../enums/view.enums';
```
Comment (evavirseda) [inline apps/wallet-dashboard/components/Dialogs/settings/hooks/useSettingsDialog.ts]:
```suggestion
import { SettingsDialogView } from '../enums';
```

File: apps/wallet-dashboard/components/Dialogs/settings/views/NetworkSelectorView.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    const { addNotification } = useNotifications();
+    const clientContext = useIotaClientContext();
+    const activeNetwork = clientContext.network;
+
+    async function handleNetworkChange(network: NetworkConfiguration) {
+        if (activeNetwork === network.id) {
+            return;
+        }
+        clientContext.selectNetwork(network.id);
+        addNotification(`Switched to ${network.name}`, NotificationType.Success);
```
Comment (evavirseda) [inline apps/wallet-dashboard/components/Dialogs/settings/views/NetworkSelectorView.tsx]: we should use `toast.success` here to have styled toasts :see_no_evil:  I'll open an issue to remove everything related to `addNotification`, `useNotifcations`,etc. but can you please update this to use toast? :pray:

**Review by evavirseda — CHANGES_REQUESTED** [review]

File: apps/wallet-dashboard/components/Dialogs/settings/views/NetworkSelectorView.tsx (hunk 2 above)
Comment (evavirseda) [reply apps/wallet-dashboard/components/Dialogs/settings/views/NetworkSelectorView.tsx]: https://github.com/iotaledger/iota/issues/4520

**Review by evavirseda — APPROVED** [review]

File: apps/wallet-dashboard/components/Dialogs/settings/enums/view.enums.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+export enum SettingsDialogView {
+    SelectSetting = 'SelectSetting',
+    NetworkSettings = 'NetworkSettings',
+    AutoLockProfile = 'AutoLockProfile',
+    FAQ = 'FAQ',
+    Themes = 'Themes',
+    Logout = 'Logout',
```
Comment (cpl121) [inline apps/wallet-dashboard/components/Dialogs/settings/enums/view.enums.ts]: we're just using `NetworkSettings` right?
I don't know if the rest are necessary, for example, we have the theme change outside the settings

**Review by cpl121 — CHANGES_REQUESTED** [review]

File: apps/wallet-dashboard/app/(protected)/components/top-nav/TopNav.tsx [hunk 4]
```diff
@@ -1,18 +1,43 @@
...
+
     return (
         <div className="flex w-full flex-row items-center justify-end gap-md py-xs--rs">
-            <Badge label="Mainnet" type={BadgeType.PrimarySoft} />
+            <Badge
+                label={networkName}
+                type={network === Network.Mainnet ? BadgeType.PrimarySoft : BadgeType.Neutral}
+            />
             <ConnectButton size="md" />
+            <Button
```
Comment (begonaalvarezd) [inline apps/wallet-dashboard/app/(protected)/components/top-nav/TopNav.tsx]: i would put the settings icon the last element

File: apps/wallet-dashboard/components/Dialogs/settings/enums/view.enums.ts (hunk 3 above)
Comment (begonaalvarezd) [reply apps/wallet-dashboard/components/Dialogs/settings/enums/view.enums.ts]: For now we can have here just `SelectSetting` & `NetworkSettings` indeed

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
beautiful work, just 2 nits

**Review by begonaalvarezd — COMMENTED** [review]
there are conflicts too 🙏🏼

**Review by begonaalvarezd — APPROVED** [review]

**Review by cpl121 — APPROVED** [review]

**Pull request merged**